jb/fftw/tde_result.hpp

```diff
@@ -0,0 +1,122 @@
+#ifndef jb_fftw_tde_result_hpp
+#define jb_fftw_tde_result_hpp
+
+#include <jb/detail/array_traits.hpp>
+#include <jb/fftw/aligned_multi_array.hpp>
+
+namespace jb {
+namespace fftw {
+
+/**
+ * tde_result is a type to handle the result of operating over the
```

Comment:
I would change that description to something like:

```
 * A time-delay estimator (TDE) is an algorithm to compare
  * two families of timeseries and return the estimated delay
  * of the first family vs. the second family.
  * When the families of timeseries are represented by an array
  * of dimension K, the last dimension is interpreted as time,
  * and the remaining K-1 dimensions are interpreted as the
  * family parameters.
  * In that case, the output of a TDE is an array of one dimension
  * less than the inputs to the TDE algorithm.
  *
  * In contrast, when the input into the TDE algorithm is a simple
  * vector, or a one dimensional array then the output is a simple value.
  *
  * This template class, and its specializations, compute the representation for
  * the TDE result given the input type.  Because the TDE algorithms usually have
  * two outputs (the time delay estimate, usually represented by an integer,
  * and the confidence, usually a floating point type), the class is parametric
  * on the output type.
  * 
```